lms/djangoapps/instructor/views/api.py

```diff
@@ -772,6 +772,49 @@ def extract_user_info(user):
 @ensure_csrf_cookie
 @cache_control(no_cache=True, no_store=True, must_revalidate=True)
 @require_level('staff')
+def get_problem_responses(request, course_id):
+    """
+    Initiate generation of a CSV file containing all student answers
+    to a given problem.
+
+    Responds with JSON
+        {"status": "... status message ..."}
+
+    if initiation is successful (or generation task is already running).
+
+    Responds with BadRequest if problem URL is faulty.
+    """
+    course_key = CourseKey.from_string(course_id)
+    problem_url = request.GET.get('problem_url', '')
+
+    if problem_url[-4:] == ".xml":
+        problem_url = problem_url[:-4]
+
+    try:
+        course_key.make_usage_key_from_deprecated_string(problem_url)
+    except InvalidKeyError:
+        problem_not_found_status = _("Could not find problem with this URL.")
+        return JsonResponseBadRequest(problem_not_found_status)
```

Comment:
Try to avoid [re]using any of the Opaque Key methods that have "deprecated" in the name.

I think the new way to write this would be something like the following, though it may need some refinement. I'm not sure of exactly what formats need to be accepted for the problem URL.

```python
try:
    problem_key = UsageKey.from_string(problem_url)
    if problem_key.course != course_key:
        raise InvalidKeyError
except InvalidKeyError:
    return JsonResponseBadRequest(_("Could not find problem with this URL."))
```